pkg/cloud/services/compute/service.go

```diff
@@ -63,8 +64,12 @@ func NewService(scope *scope.Scope) (*Service, error) {
 	volumeClient, err := openstack.NewBlockStorageV3(scope.ProviderClient, gophercloud.EndpointOpts{
 		Region: scope.ProviderClientOpts.RegionName,
 	})
+	noVolumeService := false
 	if err != nil {
-		return nil, fmt.Errorf("failed to create volume service client: %v", err)
+		// failed to create volume service client, we should tolerate
+		scope.Logger.Info("failed to create volume service client, no volume related operations allowed")
+		volumeClient = nil
```

Comment:
This isn't strictly necessary as we already set it to nil when returning an error.